azure/src/main/scala/quasar/blobstore/azure/AzureListService.scala

```diff
@@ -25,23 +25,25 @@ import java.lang.Integer
 import scala.{None, Some}
 
 import cats.data.Kleisli
-import cats.effect.{Async, ContextShift}
-import com.microsoft.azure.storage.blob.{ContainerURL, ListBlobsOptions}
-import com.microsoft.rest.v2.Context
+import cats.effect.{ConcurrentEffect, ContextShift}
+import cats.syntax.option._
+import com.azure.storage.blob.BlobContainerAsyncClient
+import com.azure.storage.blob.models.ListBlobsOptions
 
 object AzureListService {
-  def apply[F[_]: Async: ContextShift](
+
+  def apply[F[_]: ConcurrentEffect: ContextShift](
       toListBlobsOption: Kleisli[F, PrefixPath, ListBlobsOptions],
       mkArgs: ListBlobsOptions => ListBlobHierarchyArgs)
       : ListService[F] =
     toListBlobsOption map
       mkArgs andThen
       requests.listRequestK andThen
-      converters.toBlobstorePathsK
+      converters.toBlobstorePathsK.map(_.some)
```

Comment:
As discussed with @wemrysi we are deviating from the datasource spec here. The spec mandates to return `None` for non-existing paths and `Some` (empty stream) for existing non-prefix blobpaths. However we are returning `Some` empty stream for both cases here.

So no need to use `peek1` anymore. 

Created ch11270 to discuss further